index.js

```diff
@@ -1,88 +1,101 @@
-    return item.resource;
-}
-
-Model.prototype.findAll = async function() {
-    await this.initialize();
-    
-    const { resources } = await this.container.items
-        .query("SELECT * FROM docs")
-        .fetchAll();
-
-    return resources;
-}
-
-Model.prototype.find = async function(id) {
-    await this.initialize();
-    
-    const response = await this.container.item(id).read();
-    const item = response.resource;
-
-    return item;
-}
-
-Model.prototype.update = async function(id, document) {
-    await this.initialize();
-    
-    try {
-        const oldRecord = await this.find(id);
-        let updatedRecord = {...oldRecord, ...document};
-        
-        updatedRecord.id = id;
-        updatedRecord.updatedAt = new Date();
-        
-        updatedRecord = await this.container
-            .item(id)
-            .replace(updatedRecord);
-        
-        return updatedRecord;
-    }
-    catch(e) {
-        throw new Error(e);
-    }
-}
-
-Model.prototype.destroy = async function(id) {
-    await this.initialize();
-    
-    const response = await this.container.item(id).delete();
-    return response;
-}
-
-
-module.exports = Model;
\ No newline at end of file
+  return {
+    id: document.id,
+    firstname: document.firstname,
+    lastname: document.lastname,
+    createdAt: document.createdAt,
+    updatedAt: document.updatedAt,
+  };
+};
+
+Model.prototype.save = async function (document) {
+  const currentDate = new Date();
+  let item = this.formatModel(document);
+  await this.initialize();
+
+  item.createdAt = document.createdAt ? document.createdAt : currentDate;
+  item.updatedAt = document.updatedAt ? document.updatedAt : currentDate;
+
+  item = await this.container.items.create(item);
+  return item.resource;
+};
+
+Model.prototype.findAll = async function () {
+  await this.initialize();
+
+  const { resources } = await this.container.items
+    .query("SELECT * FROM docs")
+    .fetchAll();
+
+  return resources;
+};
+
+Model.prototype.find = async function (id) {
+  await this.initialize();
+
+  const response = await this.container.item(id).read();
+  const item = response.resource;
+
+  return item;
+};
+
+Model.prototype.update = async function (id, document) {
+  await this.initialize();
+
+  try {
+    const oldRecord = await this.find(id);
+    let updatedRecord = { ...oldRecord, ...document };
+
+    updatedRecord.id = id;
+    updatedRecord.updatedAt = new Date();
+
+    updatedRecord = await this.container.item(id).replace(updatedRecord);
+
+    return updatedRecord;
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+Model.prototype.destroy = async function (id) {
+  await this.initialize();
+
+  const response = await this.container.item(id).delete();
+  return response;
+};
+
+//findOne - returns an object of records that contain the value of a particular key
+Model.prototype.findOne = async function (key, value) {
+  if (Model.hasOwnProperty(!key)) return null;
```

Comment:
Not sure what you're trying to achieve here:
1. Is the intent to save the key:value on to the Model Object? If so, would you want to set that property as well once the Fetch is completeed?
2. Not sure that we need to cache the value of key here - if it's a common query perhaps there is another way to keep this persisted?